lib/core/utils/flattened-tree.js

```diff
@@ -144,26 +146,14 @@ axe.utils.getFlattenedTree = function(node, shadowId) {
 };
 
 /**
- * Recursively return a single node from a virtual dom tree
+ * Return a single node from the virtual dom tree
  *
- * @param {Object} vNode The flattened, virtual DOM tree
+ * @param {Object} vNode The flattened, virtual DOM tree (optional)
  * @param {Node}   node  The HTML DOM node
  */
 axe.utils.getNodeFromTree = function(vNode, node) {
-	var found;
-
-	if (vNode.actualNode === node) {
-		return vNode;
-	}
-	vNode.children.forEach(candidate => {
-		if (found) {
-			return;
-		}
-		if (candidate.actualNode === node) {
-			found = candidate;
-		} else {
-			found = axe.utils.getNodeFromTree(candidate, node);
-		}
-	});
-	return found;
+	// TODO: removing vNode is a breaking change, but we can skirt that
+	// for now by ignoring it if passed in
```

Comment:
I'm wondering if it isn't better to keep `getNodeFromTree` as is and deprecate it, and have a replacement method that uses cache? Would like to discuss.

Reply:
👎 from me. I see this as an opportunity for people (us) to continue using the "wrong method". Additionally, as discussed yesterday, there are 3 different methods for figuring out if something is "hidden". Let's not create the same problem here.

Reply:
This is a breaking change getNodeFromTree is something that rules can use. Why is this change necessary?